# Thomas Hoffmann, EMBL Heidelberg, stuctures-it@embl.de, 2024/05
# morphosamplers used pydantic 1, but napari 0.4.19 used pydantic 2
# enforce using _evented_models_v1 with pydantic.v1 although pydantic2 is loaded.
# In order to avoid conflicts with other modules using psygnal, rename it to psygnalV1.
diff -ru psygnal-0.9.5/pyproject.toml psygnal-0.9.5_rename_and_enforce_pydantic_v1/pyproject.toml
--- psygnal-0.9.5/pyproject.toml	2020-02-02 01:00:00.000000000 +0100
+++ psygnal-0.9.5_rename_and_enforce_pydantic_v1/pyproject.toml	2024-05-07 12:16:45.243650060 +0200
@@ -5,7 +5,7 @@
 
 # https://peps.python.org/pep-0621/
 [project]
-name = "psygnal"
+name = "psygnalV1"
 description = "Fast python callback/event system modeled after Qt Signals"
 readme = "README.md"
 requires-python = ">=3.7"
diff -ru psygnal-0.9.5/src/psygnal/containers/_evented_dict.py psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/containers/_evented_dict.py
--- psygnal-0.9.5/src/psygnal/containers/_evented_dict.py	2020-02-02 01:00:00.000000000 +0100
+++ psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/containers/_evented_dict.py	2024-05-07 12:22:00.902460253 +0200
@@ -14,8 +14,8 @@
     Union,
 )
 
-from psygnal._group import SignalGroup
-from psygnal._signal import Signal
+from psygnalV1._group import SignalGroup
+from psygnalV1._signal import Signal
 
 _K = TypeVar("_K")
 _V = TypeVar("_V")
diff -ru psygnal-0.9.5/src/psygnal/containers/_evented_list.py psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/containers/_evented_list.py
--- psygnal-0.9.5/src/psygnal/containers/_evented_list.py	2020-02-02 01:00:00.000000000 +0100
+++ psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/containers/_evented_list.py	2024-05-07 12:22:00.902460253 +0200
@@ -25,9 +25,9 @@
 
 from typing import Any, Iterable, MutableSequence, TypeVar, Union, cast, overload
 
-from psygnal._group import EmissionInfo, SignalGroup
-from psygnal._signal import Signal, SignalInstance
-from psygnal.utils import iter_signal_instances
+from psygnalV1._group import EmissionInfo, SignalGroup
+from psygnalV1._signal import Signal, SignalInstance
+from psygnalV1.utils import iter_signal_instances
 
 _T = TypeVar("_T")
 Index = Union[int, slice]
diff -ru psygnal-0.9.5/src/psygnal/containers/_evented_proxy.py psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/containers/_evented_proxy.py
--- psygnal-0.9.5/src/psygnal/containers/_evented_proxy.py	2020-02-02 01:00:00.000000000 +0100
+++ psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/containers/_evented_proxy.py	2024-05-07 12:22:00.902460253 +0200
@@ -9,8 +9,8 @@
         f"{e}. Please `pip install psygnal[proxy]` to use EventedObjectProxies"
     ) from e
 
-from psygnal._group import SignalGroup
-from psygnal._signal import Signal
+from psygnalV1._group import SignalGroup
+from psygnalV1._signal import Signal
 
 T = TypeVar("T")
 _UNSET = object()
diff -ru psygnal-0.9.5/src/psygnal/containers/_evented_set.py psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/containers/_evented_set.py
--- psygnal-0.9.5/src/psygnal/containers/_evented_set.py	2020-02-02 01:00:00.000000000 +0100
+++ psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/containers/_evented_set.py	2024-05-07 12:22:00.902460253 +0200
@@ -3,7 +3,7 @@
 from itertools import chain
 from typing import TYPE_CHECKING, Any, Iterable, Iterator, MutableSet, TypeVar
 
-from psygnal import Signal, SignalGroup
+from psygnalV1 import Signal, SignalGroup
 
 if TYPE_CHECKING:
     from typing_extensions import Final
@@ -211,7 +211,7 @@
 
     Examples
     --------
-    >>> from psygnal.containers import EventedSet
+    >>> from psygnalV1.containers import EventedSet
     >>>
     >>> my_set = EventedSet([1, 2, 3])
     >>> my_set.events.items_changed.connect(
diff -ru psygnal-0.9.5/src/psygnal/containers/_selection.py psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/containers/_selection.py
--- psygnal-0.9.5/src/psygnal/containers/_selection.py	2020-02-02 01:00:00.000000000 +0100
+++ psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/containers/_selection.py	2024-05-07 12:22:00.903460278 +0200
@@ -2,7 +2,7 @@
 
 from typing import TYPE_CHECKING, Any, Container, TypeVar
 
-from psygnal._signal import Signal
+from psygnalV1._signal import Signal
 
 from ._evented_set import BailType, EventedOrderedSet, SetEvents
 
diff -ru psygnal-0.9.5/src/psygnal/_evented_decorator.py psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/_evented_decorator.py
--- psygnal-0.9.5/src/psygnal/_evented_decorator.py	2020-02-02 01:00:00.000000000 +0100
+++ psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/_evented_decorator.py	2024-05-07 12:22:00.903460278 +0200
@@ -10,7 +10,7 @@
     overload,
 )
 
-from psygnal._group_descriptor import SignalGroupDescriptor
+from psygnalV1._group_descriptor import SignalGroupDescriptor
 
 if TYPE_CHECKING:
     from typing_extensions import Literal
@@ -109,7 +109,7 @@
     Examples
     --------
     ```python
-    from psygnal import evented
+    from psygnalV1 import evented
     from dataclasses import dataclass
 
     @evented
diff -ru psygnal-0.9.5/src/psygnal/_evented_model_v1.py psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/_evented_model_v1.py
--- psygnal-0.9.5/src/psygnal/_evented_model_v1.py	2020-02-02 01:00:00.000000000 +0100
+++ psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/_evented_model_v1.py	2024-05-07 12:53:04.900944241 +0200
@@ -30,9 +30,9 @@
     EqOperator = Callable[[Any, Any], bool]
 
 else:
-    import pydantic.main as pydantic_main
-    from pydantic import BaseModel, PrivateAttr, utils
-    from pydantic.fields import Field, FieldInfo
+    import pydantic.v1.main as pydantic_main
+    from pydantic.v1 import BaseModel, PrivateAttr, utils
+    from pydantic.v1.fields import Field, FieldInfo
 
     def dataclass_transform(*args, **kwargs):
         return lambda a: a
diff -ru psygnal-0.9.5/src/psygnal/_group_descriptor.py psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/_group_descriptor.py
--- psygnal-0.9.5/src/psygnal/_group_descriptor.py	2020-02-02 01:00:00.000000000 +0100
+++ psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/_group_descriptor.py	2024-05-07 12:22:00.903460278 +0200
@@ -163,7 +163,7 @@
     Examples
     --------
     ```python
-    from psygnal import evented, get_evented_namespace, is_evented
+    from psygnalV1 import evented, get_evented_namespace, is_evented
 
     @evented(events_namespace="my_events")
     class Foo:
@@ -283,8 +283,8 @@
         an event.  Or you can use `evented_setattr` yourself
 
         ```python
-        from psygnal._group_descriptor import evented_setattr
-        from psygnal import SignalGroupDescriptor
+        from psygnalV1._group_descriptor import evented_setattr
+        from psygnalV1 import SignalGroupDescriptor
         from dataclasses import dataclass
         from typing import ClassVar
 
@@ -333,7 +333,7 @@
     ```python
     from typing import ClassVar
     from dataclasses import dataclass
-    from psygnal import SignalGroupDescriptor
+    from psygnalV1 import SignalGroupDescriptor
 
     @dataclass
     class Person:
diff -ru psygnal-0.9.5/src/psygnal/_group.py psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/_group.py
--- psygnal-0.9.5/src/psygnal/_group.py	2020-02-02 01:00:00.000000000 +0100
+++ psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/_group.py	2024-05-07 12:22:00.903460278 +0200
@@ -21,7 +21,7 @@
 
 from mypy_extensions import mypyc_attr
 
-from psygnal._signal import Signal, SignalInstance, _SignalBlocker
+from psygnalV1._signal import Signal, SignalInstance, _SignalBlocker
 
 __all__ = ["EmissionInfo", "SignalGroup"]
 
diff -ru psygnal-0.9.5/src/psygnal/__init__.py psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/__init__.py
--- psygnal-0.9.5/src/psygnal/__init__.py	2020-02-02 01:00:00.000000000 +0100
+++ psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/__init__.py	2024-05-07 12:20:52.304759479 +0200
@@ -6,7 +6,7 @@
 import os
 from typing import TYPE_CHECKING, Any
 
-if TYPE_CHECKING:
+if TYPE_CHECKING or True:
     PackageNotFoundError = Exception
     from ._evented_model_v1 import EventedModel  # noqa: TCH004
 
@@ -22,7 +22,7 @@
 
 
 try:
-    __version__ = version("psygnal")
+    __version__ = version("psygnalV1")
 except PackageNotFoundError:  # pragma: no cover
     __version__ = "0.0.0"
 __author__ = "Talley Lambert"
@@ -73,12 +73,9 @@
 
 def __getattr__(name: str) -> Any:
     if name == "EventedModel":
-        import pydantic.version
+        import pydantic.v1.version
 
-        if pydantic.version.VERSION.startswith("2"):
-            from ._evented_model_v2 import EventedModel
-        else:
-            from ._evented_model_v1 import EventedModel  # type: ignore
+        from ._evented_model_v1 import EventedModel  # type: ignore
 
         return EventedModel
     raise AttributeError(  # pragma: no cover
diff -ru psygnal-0.9.5/src/psygnal/_signal.py psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/_signal.py
--- psygnal-0.9.5/src/psygnal/_signal.py	2020-02-02 01:00:00.000000000 +0100
+++ psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/_signal.py	2024-05-07 12:22:00.904460303 +0200
@@ -59,7 +59,7 @@
     provided in the constructor:
 
     ```python
-    from psygnal import Signal
+    from psygnalV1 import Signal
 
     class MyEmitter:
         changed = Signal(int)
@@ -207,7 +207,7 @@
         Examples
         --------
         ```python
-        from psygnal import Signal
+        from psygnalV1 import Signal
 
         def my_callback():
             source = Signal.current_emitter()
diff -ru psygnal-0.9.5/src/psygnal/_throttler.py psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/_throttler.py
--- psygnal-0.9.5/src/psygnal/_throttler.py	2020-02-02 01:00:00.000000000 +0100
+++ psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/_throttler.py	2024-05-07 12:22:00.904460303 +0200
@@ -181,7 +181,7 @@
     Examples
     --------
     ```python
-    from psygnal import Signal, throttled
+    from psygnalV1 import Signal, throttled
 
     class MyEmitter:
         changed = Signal(int)
@@ -237,7 +237,7 @@
     Examples
     --------
     ```python
-    from psygnal import Signal, debounced
+    from psygnalV1 import Signal, debounced
 
     class MyEmitter:
         changed = Signal(int)
diff -ru psygnal-0.9.5/src/psygnal/_weak_callback.py psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/_weak_callback.py
--- psygnal-0.9.5/src/psygnal/_weak_callback.py	2020-02-02 01:00:00.000000000 +0100
+++ psygnal-0.9.5_rename_and_enforce_pydantic_v1/src/psygnal/_weak_callback.py	2024-05-07 12:22:00.904460303 +0200
@@ -82,7 +82,7 @@
     Examples
     --------
     ```python
-        from psygnal._weak_callback import weak_callback
+        from psygnalV1._weak_callback import weak_callback
 
     class T:
         def greet(self, name):
Only in psygnal-0.9.5_rename_and_enforce_pydantic_v1/src: psygnalV1
